base_decorator.rb

```diff
@@ -0,0 +1,16 @@
+require_relative 'nameable'
+
+# BaseDecorator is a base class for all decorators.
+# It implements the Nameable interface and delegates the correct_name method to the wrapped object.
+class BaseDecorator < Nameable
+  # Initializes a new BaseDecorator with a nameable object to be wrapped.
+  def initialize(nameable, *args)
+    super(*args)
+    @nameable = nameable
+  end
```

Comment:
- [OPTIONAL] Great work refactoring your code and making it work. But you can make it simple and less ambiguous by calling the `super()` without the `*args`.

```ruby
class BaseDecorator < Nameable
  # Initializes a new BaseDecorator with a nameable object to be wrapped.
  def initialize(nameable)
    super()
    @nameable = nameable
  end
  ...
end
```